docs/DeveloperGuide.md

```diff
@@ -1,38 +1,253 @@
 # Developer Guide
 
-## Acknowledgements
+## Table of Contents
+
+1. [Acknowledgements](#acknowledgements)
+2. [Design](#design)
+   - [Architecture](#architecture)
+   - [UI component](#ui-component)
+   - [Parser component](#parser-component)
+   - [Storage component](#storage-component)
+   - [Common classes](#common-classes)
+3. [Implementation](#implementation)
+4. [Appendix: Requirements](#appendix--requirements)
+5. [Appendix: Instructions for manual testing](#appendix--instructions-for-manual-testing)
+
+---
 
+## Acknowledgements
+x
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
+{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the
+original source as well}
+
+- [addressbook-level2](https://github.com/se-edu/addressbook-level2)
+- [addressbook-level3](https://github.com/se-edu/addressbook-level3)
+
+---
+
+## Design
+
+### Architecture
+
+![Architecture Diagram](images/ArchitectureDiagram.png)
+
+The ***architecture diagram*** given above is explains the high level design of the program.
+
+Given below is a quick overview of the main components and how they interact with each other.
+
+**Main components of architecture**
+
+`BrokeMan` has one class [`Main`](https://github.com/AY2223S2-CS2113-F13-2/tp/blob/master/src/main/java/seedu/brokeMan/BrokeMan.java), which is responsible for:
+- At program launch: Initialises the components in the correct sequence, and connect them up with each other
+- At program termination: Shuts down the components and invokes cleanup methods where necessary.
+
+[`Common`](#common-class) represents a collection of messages used by multiple other components.
```

Comment:
![image](https://user-images.githubusercontent.com/88487773/228703886-4b2fd0c7-095e-4de3-9560-4cfa7de2110b.png)
A little confused with the architecture diagram, in line 46 it mentions that "common represents a collection of messages used by multiple other components, but there seems to be no arrow associated with the "common" component in the diagram, is it supposed to be intended this way?